com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/topology.py

```diff
@@ -232,15 +234,43 @@ def parallel(self, width):
         
         Args:
             width (int): degree of parallelism
+            routing - denotes what type of tuple routing to use. 
+                ROUND_ROBIN: delivers tuples in round robin fashion to downstream operators
+                HASH_PARTIONED: delivers to downstream operators based on the hash of the tuples being sent
+                or if a function is provided the function will be called to provide the hash
+            func - Optional function called when HASH_PARTIONED routing is specified.  The function provides an
+                int32 value to be used as the hash that determines the tuple routing to downstream operators
+
         Returns:
             Stream
+
         """
-        iop = self.isolate()
-               
-        op2 = self.topology.graph.addOperator("$Parallel$")
-        op2.addInputPort(outputPort=iop.getOport())
-        oport = op2.addOutputPort(width)
-        return Stream(self.topology, oport)
+        if (routing == None or routing == Routing.ROUND_ROBIN) :
+            iop = self.isolate()                  
+            op2 = self.topology.graph.addOperator("$Parallel$")
+            op2.addInputPort(outputPort=iop.getOport())
+            oport = op2.addOutputPort(width)
+            return Stream(self.topology, oport)
+        elif(routing == Routing.HASH_PARTITIONED ) :
+            if (func is None) :
+                op = self.topology.graph.addOperator("com.ibm.streamsx.topology.functional.python::PyFunctionHashAdder", hash)
```

Comment:
Minor point but this can be simplified to just set func to hash when func is None and then no need to repeat the addOperator line.